scripts/finetune.py

```diff
@@ -279,7 +358,14 @@ def train(
                 prompter = None
             else:
                 prompter = kwargs["prompter"]
-        do_inference(cfg, model, tokenizer, prompter=prompter)
+
+        if cfg.server:
+            server_address = (cfg.server_addr, cfg.server_port)
+            httpd = socketserver.TCPServer(server_address, lambda *args, **kwargs: HttpHandler(*args, cfg=cfg, prompter=prompter, tokenizer=tokenizer, model=model, **kwargs))
+            print(f"Server running on port {cfg.server_port}")
+            httpd.serve_forever()
```

Comment:
should this be explicitly killed at the end of training?

Reply:
Can you train and run inference at the same time?  This only runs if you're in --inference mode.